lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -1652,6 +1649,35 @@ AddressMaterialization::materializeProjectionIntoUse(Operand *operand,
   }
 }
 
+// Updates \p storage's address to the dominating instruction of \p address and
+// its old value (if any).
+//
+// If the old address is replaced, its uses are replaced with the new address.
+//
+// The dominated instruction is deleted.
```

Comment:
This comment could be even more explicit about the hard requirement that we only materialize one of the address projections! Otherwise, the memory verify will break on init_enum_data_addr